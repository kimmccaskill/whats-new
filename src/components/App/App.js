import React, { Component } from 'react';
import entertainment from '../../data/entertainment';
import health from '../../data/health';
import local from '../../data/local';
import science from '../../data/science';
import technology from '../../data/technology';
import './App.css';
import NewsContainer from '../NewsContainer/NewsContainer.js'
import Menu from '../Menu/Menu.js'

class App extends Component {
  constructor() {
    super();
    this.state = {
      categories: {
        Entertainment : {data: entertainment},
        Health: {data: health},
        Local: {data: local},
        Science: {data: science},
        Technology: {data: technology}},
      articleShown: {data: local}
    }
  }

  changeCategory = (category) => {
    this.setState({articleShown: {data: this.state.categories[category].data}})
  }

  render () {
    return (
      <div className="app">
        <Menu changeCategory={this.changeCategory} data={this.state}/>
        <NewsContainer news={this.state.articleShown.data}/>
      </div>
    );
  }
}

export default App;
