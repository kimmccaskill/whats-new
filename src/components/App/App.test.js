import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import entertainment from '../../data/entertainment'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  let wrapper;
  const mockChangeCategory = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state of no ideas', () => {
  expect(wrapper.state('categories')).toEqual({})
  expect(wrapper.state('articleShown')).toEqual({})
  });

  it('should update state with changeCategory method', () => {
  const mockEvent = {
    target: {
      value: 'Entertainment'
    }
  }
  const expectedArticleShown = {data: entertainment};

  wrapper.instance().changeCategory(mockEvent);

  expect(expectedArticleShown.data).toEqual(entertainment);
  });

  it('should update state of articleShown with filterArticles method', () => {
    const mockQuery = 'cancel'
    wrapper.instance().filterArticles = jest.fn()

    wrapper.instance().filterArticles(mockQuery);
    expect(wrapper.instance().filterArticles).toHaveBeenCalled();
  });

});
