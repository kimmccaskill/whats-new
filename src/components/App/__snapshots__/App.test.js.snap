// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="app"
>
  <Menu
    changeCategory={[Function]}
    data={
      Object {
        "articleShown": Object {},
        "categories": Object {},
      }
    }
  />
  <section
    className="main-section"
  >
    <SearchForm
      filterArticles={[Function]}
    />
    <NewsContainer />
  </section>
</main>
`;
